src/analysis/retail/demonhunter/vengeance/modules/Abilities.tsx

```diff
@@ -259,7 +270,19 @@ class Abilities extends CoreAbilities {
       {
         spell: SPELLS.INFERNAL_STRIKE.id,
         category: SPELL_CATEGORY.UTILITY,
-        cooldown: 20,
+        cooldown: () => {
+          const baseCooldown = 20;
+          const abyssalHasteReduction =
+            ABYSSAL_HASTE_SCALING[
+              combatant.getTalentRank(TALENTS_DEMON_HUNTER.ABYSSAL_HASTE_VENGEANCE_TALENT)
+            ];
+          const erraticFelheartReduction =
+            ERRATIC_FELHEART_SCALING[
+              combatant.getTalentRank(TALENTS_DEMON_HUNTER.ERRATIC_FELHEART_TALENT)
+            ];
+          const flatReduced = baseCooldown - abyssalHasteReduction;
+          return flatReduced - flatReduced * erraticFelheartReduction;
+        },
```

Comment:
this is complex enough that you might want to move it to be an exported function in the `InfernalStrike` file.